src/data/finvec.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2020 Reid Barton. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Reid Barton
+-/
+
+import data.equiv.fin
+
+/-!
+# Finite vectors, implemented as functions on `fin n`.
+
+The main advantage of `finvec` over alternatives like `vector` is that
```

Comment:
IMHO the only advantage over `vector` is that you can use `finvec` in nested inductives:
```lean
inductive term
| mk (f : string) (n : ℕ) (args : finvec n term) : term
```

Reply:
I certainly don't agree this is the only advantage. In math an n-tuple is effectively a function defined on the set {1, ..., n}. Modeling it by a `vector` is like using a function type without definitional beta/eta rules. For example, imagine defining coordinate-wise addition on `vector real n` and proving it is commutative versus using `finvec n real` (or `fin n -> real`).